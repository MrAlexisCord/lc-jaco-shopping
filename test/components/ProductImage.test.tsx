import React from 'react';
import renderer from 'react-test-renderer';
import { ProductCard } from '../../src/components/ProductCard';
import { ProductImage } from '../../src/components/ProductImage';
import { product2 } from '../data/products';

describe('TESTs on ProductImage', () => {

    test('***   Debe mostrar la imagen personallizada', () => {
        const wrapper = renderer.create(
            <ProductImage img='https://hola.png' />
        )

        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    test('***   Debe mostrar el component con la imagen del producto (from context)', () => {
        const wrapper = renderer.create(
            <ProductCard product={product2}>{
                () => (<ProductImage />)
            }
            </ProductCard>
        )

        expect(wrapper.toJSON()).toMatchSnapshot();
    });
})
