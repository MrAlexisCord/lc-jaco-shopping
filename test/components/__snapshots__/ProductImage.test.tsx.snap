// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TESTs on ProductImage ***   Debe mostrar el component con la imagen del producto (from context) 1`] = `
<div
  className="productCard "
>
  <img
    alt="Product"
    className="productImg "
    src="./coffee-mug.png"
  />
</div>
`;

exports[`TESTs on ProductImage ***   Debe mostrar la imagen personallizada 1`] = `
<img
  alt="Product"
  className="productImg "
  src="https://hola.png"
/>
`;
